### DISCUSSION TITLE
corrupt beets db: empy albums

### DISCUSSION THREAD

**Discussion by odoublewen** [beetbox/beets#5525] [General]
Somehow, I've managed to create two "empty" albums in my beets db.  Note that these are self-produced and self-tagged mp3 files.  I may have tagged, and retagged, and imported them a couple times before I got everything right.  Somehow, in the process of doing that, I've created two "empty" album IDs.  In other words, I think the beets database is corrupt.

Here, I have two actual mp3 tracks associated with an album "Vernal Pools". 
```
beet ls -f '$id - $album_id - $album' Vernal Pools
65174 - 4693 - Vernal Pools
65175 - 4693 - Vernal Pools
```

If I do a `ls -a` I see this:
```
$ beet ls -a -f '$id - $album' Vernal Pools
4672 - Vernal Pools
4679 - Vernal Pools
4693 - Vernal Pools
```

The problem is that album IDs 4672 and 4679 actually don't exist.  For example, if I add `$path` to the output format, I get this:
```
$ beet ls -a -f '$id - $album - $path' Vernal Pools
Traceback (most recent call last):
...clip...
  File "/home/beets2/lib/python3.9/site-packages/beets/library.py", line 1409, in item_dir
    raise ValueError("empty album for album id %d" % self.id)
ValueError: empty album for album id 4672
```

I look at [line 1409 in the library module](https://github.com/beetbox/beets/blob/master/beets/library.py#L1409C31-L1409C52) and I scratch my head, wondering, is this exception caught somewhere else, or is it simply there to provide a more descriptive error message?

What can I do to fix this?  From what I can tell from reading the docs and looking at the CLI, it doesn't seem that this is a common thing.  Note that I cannot use the `rm` subcommand to remove the album:

```
$ beet rm album_id:4672
error: No matching items found.
```

 It seems like I will need to get into the sqlite database and manually fix it by deleting these two album rows.  Can anyone provide any tips?

**Comment by odoublewen** [comment]
Apparently, this situation is the result of me using `--set` to set the album_id at time of import.   I now think this is a bug.  See https://github.com/beetbox/beets/issues/5012 for more info.

**Comment by arsaboo** [comment]
Can you try using beet update to update the db?

**Reply by odoublewen** [reply]
Thanks for the suggestion.  Unfortunately `beet update` doesn't seem to have any effect.
```
$ beet update
$ beet ls -a -p
Traceback (most recent call last):
...
  File "/home/beets2/lib/python3.9/site-packages/beets/library.py", line 1409, in item_dir
    raise ValueError("empty album for album id %d" % self.id)
ValueError: empty album for album id 2
```

The same ValueError also pops up if I do `beet info -a`

**Reply by JOJ0** [reply]
> i hate your app

Thanks for letting us know 💪
